{"tags": ["Python2.6", "pyenv2014705", "pyenv-virtualenv20140705", "OSX10.9.4"], "context": "\u4e0b\u8a18\u60c5\u5831\u306f\u53e4\u304f\u306a\u3063\u3066\u304a\u308a\u3001\u73fe\u72b6\u3067\u306f\u5b9f\u884c\u51fa\u6765\u307e\u305b\u3093\u3002\n\u8a73\u3057\u304f\u306f\u3053\u3061\u3089\u3092\u3054\u53c2\u7167\u4e0b\u3055\u3044\u307e\u305b\u3002\nPip | \u30aa\u30fc\u30d7\u30f3\u30c7\u30fc\u30bf\u3068\u30d7\u30ed\u30b0\u30e9\u30df\u30f3\u30b0\n\u7d50\u5c40PIL\u306f\u3082\u3046pip\u3067\u306f\u5165\u308c\u3089\u308c\u7121\u3055\u305d\u3046\u3067\u3059\u3002\npip\u306e\u3069\u3053\u304b\u306e\u30d0\u30fc\u30b8\u30e7\u30f3\u304b\u3089 --allow-external \u3068 --allow-unverified \u304c deprecated \u306b\u306a\u3063\u305f\u3088\u3046\u3067\u3059\u3002\n\u304a\u3068\u306a\u3057\u304f pillow \u3092\u4f7f\u3048\u3063\u3066\u4e8b\u3067\u3059\u3088\u306d\u3002\n\npyenv\u3067\u74b0\u5883\u8a2d\u5b9a\u3057\u3066\u3044\u3066\u3001pyenv local hogehoge\u3057\u3066\u3066\u3001pyenv activate\u3057\u3066\u308b\u30d7\u30ed\u30b8\u30a7\u30af\u30c8\u3067\u3001pip install PIL\u304c\u30b3\u30b1\u308b\u8a33\u3067\u3059\u3088\u3002\n\u66f4\u306b\u4e8c\u6bb5\u4ed5\u639b\u3051\u3060\u3063\u305f\u306e\u3067\u30e1\u30e2\n\n\u7d50\u8ad6\n\n\npip install PIL --allow-external PIL --allow-unverified PIL \u3067\u5165\u308c\u308b\n\nln -s /usr/local/include/freetype2 /usr/local/include/freetype \u3059\u308b\n\n\n\u8a73\u7d30\n\n1\u3064\u3081\u306e --allow-external \u306e\u4ef6\n$ pip install PIL\n\n\u3066\u3059\u308b\u3068\u30b3\u30b1\u308b\nDownloading/unpacking PIL\n  Could not find any downloads that satisfy the requirement PIL\n  Some insecure and unverifiable files were ignored (use --allow-unverified PIL to allow).\nCleaning up...\nNo distributions at all found for PIL\nStoring debug log for failure in /Users/jasper/.pip/pip.log\n\n\u30b0\u30b0\u308b\n\u3053\u3053\u306b\u66f8\u3044\u3066\u3042\u3063\u305f\nhttp://stackoverflow.com/questions/21242107/pip-install-pil-dont-install-into-virtualenv\npip install PIL --allow-external PIL --allow-unverified PIL\n\n\u3063\u3066\u3084\u3063\u305f\u3089\u306a\u306b\u3084\u3089\u307e\u3048\u3068\u9055\u3046\u611f\u3058\u306b\u306a\u3063\u305f\n\n2\u3064\u3081\u306efreetype\u306e\u4ef6\npip install PIL --allow-external PIL --allow-unverified PIL\n\n\u3067\u9032\u5c55\u3057\u305f\u306e\u306f\u3044\u3044\u3051\u3069\u307e\u3060\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u51fa\u6765\u306a\u3044\nbuilding '_imagingft' extension\n\ngcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/System/Library/Frameworks/Tcl.framework/Headers -I/System/Library/Frameworks/Tk.framework/Headers -I/usr/local/include/freetype2 -IlibImaging -I/usr/local/opt/pyenv/versions/inblue/include -I/usr/local/include -I/usr/include -I/usr/local/opt/pyenv/versions/2.6.8/include/python2.6 -c _imagingft.c -o build/temp.macosx-10.4-x86_64-2.6/_imagingft.o\n\n_imagingft.c:73:10: fatal error: 'freetype/fterrors.h' file not found\n\n#include <freetype/fterrors.h>\n\n         ^\n\n1 error generated.\n\nerror: command 'gcc' failed with exit status 1\n\n----------------------------------------\nCleaning up...\nCommand /usr/local/opt/pyenv/versions/inblue/bin/python -c \"import setuptools, tokenize;__file__='/usr/local/var/lib/pyenv/versions/inblue/build/PIL/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\n', '\n'), __file__, 'exec'))\" install --record /var/folders/10/1gcb6yqd4m14g9qr7s0lztg40000gn/T/pip-0bcmkN-record/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/opt/pyenv/versions/inblue/include/site/python2.6 failed with error code 1 in /usr/local/var/lib/pyenv/versions/inblue/build/PIL\nStoring debug log for failure in /Users/jasper/.pip/pip.log\n\n\u308f\u304b\u3093\u306a\u3044\u3002\u30b0\u30b0\u308b\n\u3053\u308c\u306b\u5f53\u305f\u308b\u3051\u3069\u9055\u3046\u3002Pillow\u3092\u5165\u308c\u308d\u3063\u3066\u3002\nhttp://stackoverflow.com/questions/23996756/pil-installation-error-on-mac-os-maverics-inside-virtual-env\n\u4ffa\u304c\u5165\u308c\u305f\u3044\u306e\u306fPIL\u306a\u3093\u3060\u3063\n\u3055\u3089\u306b\u30b0\u30b0\u308b\n\u305d\u308c\u3063\u307d\u3044\u306e\u306b\u5f53\u305f\u308b\u3002\nhttp://stackoverflow.com/questions/20325473/error-installing-python-image-library-using-pip-on-mac-os-x-10-9\n\u30b7\u30f3\u30dc\u30ea\u30c3\u30af\u30ea\u30f3\u30af\u8cbc\u3063\u3066\u308b\u3060\u3051\u3060\u3057\u3001\u30ea\u30f3\u30af\u5148\u306e\u30d5\u30a1\u30a4\u30eb\u306f\u7121\u304b\u3063\u305f\u304b\u3089\u3001\u3084\u3063\u305f\u3060\u3051\u306a\u3089\u623b\u305b\u308b\u3060\u308d\u3046\u3068\u601d\u3044\u307e\u3057\u3066\u3002\nln -s /usr/local/include/freetype2 /usr/local/include/freetype\n\n\u306f\u3044\u3002\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u6210\u529f\u3002\n\npip install PIL --allow-external PIL --allow-unverified PIL\n\n\n2014/11/20 \u8ffd\u8a18:\n@shitsyndrome \u3055\u3093\u304b\u3089\u30b3\u30e1\u30f3\u30c8\u9802\u304d\u307e\u3057\u3066\u539f\u56e0\u304c\u767a\u899a\u3057\u307e\u3057\u305f\u3002\n\npip\u306e1.5\u304b\u3089\u306f\u30c7\u30d5\u30a9\u30eb\u30c8\u306e\u6319\u52d5\u304c\u304b\u308f\u3063\u3066\u3044\u3066\u3001\u5916\u90e8\u30db\u30b9\u30c8(pypi.python.org\u4ee5\u5916)\u304b\u3089\u306e\u30c0\u30a6\u30f3\u30ed\u30fc\u30c9\u3092\u884c\u3044\u305f\u3044\u5834\u5408\u306f\u660e\u793a\u7684\u306b\u6307\u5b9a\u3059\u308b\u5fc5\u8981\u304c\u3042\u308b\u307f\u305f\u3044\u3067\u3059\u3002\nhttp://pip.readthedocs.org/en/latest/reference/pip_install.html#externally-hosted-files\n\n**\u4e0b\u8a18\u60c5\u5831\u306f\u53e4\u304f\u306a\u3063\u3066\u304a\u308a\u3001\u73fe\u72b6\u3067\u306f\u5b9f\u884c\u51fa\u6765\u307e\u305b\u3093\u3002**\n\n\u8a73\u3057\u304f\u306f\u3053\u3061\u3089\u3092\u3054\u53c2\u7167\u4e0b\u3055\u3044\u307e\u305b\u3002\n[Pip \\| \u30aa\u30fc\u30d7\u30f3\u30c7\u30fc\u30bf\u3068\u30d7\u30ed\u30b0\u30e9\u30df\u30f3\u30b0](http://opendata.jp.net/?tag=pip)\n\n\u7d50\u5c40PIL\u306f\u3082\u3046pip\u3067\u306f\u5165\u308c\u3089\u308c\u7121\u3055\u305d\u3046\u3067\u3059\u3002\n\npip\u306e\u3069\u3053\u304b\u306e\u30d0\u30fc\u30b8\u30e7\u30f3\u304b\u3089 `--allow-external` \u3068 `--allow-unverified` \u304c `deprecated` \u306b\u306a\u3063\u305f\u3088\u3046\u3067\u3059\u3002\n\n\u304a\u3068\u306a\u3057\u304f pillow \u3092\u4f7f\u3048\u3063\u3066\u4e8b\u3067\u3059\u3088\u306d\u3002\n\n------------------------------------\n\n```pyenv```\u3067\u74b0\u5883\u8a2d\u5b9a\u3057\u3066\u3044\u3066\u3001```pyenv local hogehoge```\u3057\u3066\u3066\u3001```pyenv activate```\u3057\u3066\u308b\u30d7\u30ed\u30b8\u30a7\u30af\u30c8\u3067\u3001```pip install PIL```\u304c\u30b3\u30b1\u308b\u8a33\u3067\u3059\u3088\u3002\n\n\u66f4\u306b\u4e8c\u6bb5\u4ed5\u639b\u3051\u3060\u3063\u305f\u306e\u3067\u30e1\u30e2\n\n## \u7d50\u8ad6\n* ```pip install PIL --allow-external PIL --allow-unverified PIL``` \u3067\u5165\u308c\u308b\n* ```ln -s /usr/local/include/freetype2 /usr/local/include/freetype``` \u3059\u308b\n\n## \u8a73\u7d30\n\n### 1\u3064\u3081\u306e --allow-external \u306e\u4ef6\n```shell-session\n$ pip install PIL\n```\n\u3066\u3059\u308b\u3068\u30b3\u30b1\u308b\n\n```\nDownloading/unpacking PIL\n  Could not find any downloads that satisfy the requirement PIL\n  Some insecure and unverifiable files were ignored (use --allow-unverified PIL to allow).\nCleaning up...\nNo distributions at all found for PIL\nStoring debug log for failure in /Users/jasper/.pip/pip.log\n```\n\n\u30b0\u30b0\u308b\n\n\u3053\u3053\u306b\u66f8\u3044\u3066\u3042\u3063\u305f\nhttp://stackoverflow.com/questions/21242107/pip-install-pil-dont-install-into-virtualenv\n\n```\npip install PIL --allow-external PIL --allow-unverified PIL\n```\n\u3063\u3066\u3084\u3063\u305f\u3089\u306a\u306b\u3084\u3089\u307e\u3048\u3068\u9055\u3046\u611f\u3058\u306b\u306a\u3063\u305f\n\n### 2\u3064\u3081\u306efreetype\u306e\u4ef6\n\n```\npip install PIL --allow-external PIL --allow-unverified PIL\n```\n\u3067\u9032\u5c55\u3057\u305f\u306e\u306f\u3044\u3044\u3051\u3069\u307e\u3060\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u51fa\u6765\u306a\u3044\n\n```\nbuilding '_imagingft' extension\n\ngcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/System/Library/Frameworks/Tcl.framework/Headers -I/System/Library/Frameworks/Tk.framework/Headers -I/usr/local/include/freetype2 -IlibImaging -I/usr/local/opt/pyenv/versions/inblue/include -I/usr/local/include -I/usr/include -I/usr/local/opt/pyenv/versions/2.6.8/include/python2.6 -c _imagingft.c -o build/temp.macosx-10.4-x86_64-2.6/_imagingft.o\n\n_imagingft.c:73:10: fatal error: 'freetype/fterrors.h' file not found\n\n#include <freetype/fterrors.h>\n\n         ^\n\n1 error generated.\n\nerror: command 'gcc' failed with exit status 1\n\n----------------------------------------\nCleaning up...\nCommand /usr/local/opt/pyenv/versions/inblue/bin/python -c \"import setuptools, tokenize;__file__='/usr/local/var/lib/pyenv/versions/inblue/build/PIL/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\n', '\n'), __file__, 'exec'))\" install --record /var/folders/10/1gcb6yqd4m14g9qr7s0lztg40000gn/T/pip-0bcmkN-record/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/opt/pyenv/versions/inblue/include/site/python2.6 failed with error code 1 in /usr/local/var/lib/pyenv/versions/inblue/build/PIL\nStoring debug log for failure in /Users/jasper/.pip/pip.log\n```\n\n\u308f\u304b\u3093\u306a\u3044\u3002\u30b0\u30b0\u308b\n\n\u3053\u308c\u306b\u5f53\u305f\u308b\u3051\u3069\u9055\u3046\u3002[Pillow](https://github.com/python-pillow/Pillow)\u3092\u5165\u308c\u308d\u3063\u3066\u3002\nhttp://stackoverflow.com/questions/23996756/pil-installation-error-on-mac-os-maverics-inside-virtual-env\n\n\u4ffa\u304c\u5165\u308c\u305f\u3044\u306e\u306fPIL\u306a\u3093\u3060\u3063\n\n\u3055\u3089\u306b\u30b0\u30b0\u308b\n\n\u305d\u308c\u3063\u307d\u3044\u306e\u306b\u5f53\u305f\u308b\u3002\nhttp://stackoverflow.com/questions/20325473/error-installing-python-image-library-using-pip-on-mac-os-x-10-9\n\n\u30b7\u30f3\u30dc\u30ea\u30c3\u30af\u30ea\u30f3\u30af\u8cbc\u3063\u3066\u308b\u3060\u3051\u3060\u3057\u3001\u30ea\u30f3\u30af\u5148\u306e\u30d5\u30a1\u30a4\u30eb\u306f\u7121\u304b\u3063\u305f\u304b\u3089\u3001\u3084\u3063\u305f\u3060\u3051\u306a\u3089\u623b\u305b\u308b\u3060\u308d\u3046\u3068\u601d\u3044\u307e\u3057\u3066\u3002\n\n```shell-session\nln -s /usr/local/include/freetype2 /usr/local/include/freetype\n```\n\n\u306f\u3044\u3002\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u6210\u529f\u3002\n```\npip install PIL --allow-external PIL --allow-unverified PIL\n```\n\n------------\n\n2014/11/20 \u8ffd\u8a18:\n@shitsyndrome \u3055\u3093\u304b\u3089\u30b3\u30e1\u30f3\u30c8\u9802\u304d\u307e\u3057\u3066\u539f\u56e0\u304c\u767a\u899a\u3057\u307e\u3057\u305f\u3002\n> pip\u306e1.5\u304b\u3089\u306f\u30c7\u30d5\u30a9\u30eb\u30c8\u306e\u6319\u52d5\u304c\u304b\u308f\u3063\u3066\u3044\u3066\u3001\u5916\u90e8\u30db\u30b9\u30c8(pypi.python.org\u4ee5\u5916)\u304b\u3089\u306e\u30c0\u30a6\u30f3\u30ed\u30fc\u30c9\u3092\u884c\u3044\u305f\u3044\u5834\u5408\u306f\u660e\u793a\u7684\u306b\u6307\u5b9a\u3059\u308b\u5fc5\u8981\u304c\u3042\u308b\u307f\u305f\u3044\u3067\u3059\u3002\n> http://pip.readthedocs.org/en/latest/reference/pip_install.html#externally-hosted-files\n"}